services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/grid/PatchRowHandler.java

```diff
@@ -172,10 +172,12 @@ private NewObject getSynapseRowMetadata(Row row) {
 			synapseRowMetadataObjectMap.put("etag",  etagConst.getOperationId());
 		}
 
-		// fill the `synapseRow` object
-		currentPatch.addNewOperation(InsertObject.class)
-				.setObjectId(synapseRowMetadata.getOperationId())
-				.setMap(synapseRowMetadataObjectMap);
+		if (!synapseRowMetadataObjectMap.isEmpty()) {
+			// fill the `synapseRow` object
+			currentPatch.addNewOperation(InsertObject.class)
+					.setObjectId(synapseRowMetadata.getOperationId())
+					.setMap(synapseRowMetadataObjectMap);
+		}
```

Comment:
Check if the map is empty before inserting.

Ideally, we would not need this check, and the `Patch` object would just treat this as a no-op, but that would be a substantial refactor.